src/github.com/matrix-org/dendrite/roomserver/query/query.go

```diff
@@ -40,13 +40,31 @@ type RoomserverQueryAPIDatabase interface {
 	// Look up the numeric IDs for a list of events.
 	// Returns an error if there was a problem talking to the database.
 	EventNIDs(eventIDs []string) (map[string]types.EventNID, error)
-	// Look up the join events for all members in a room as requested by a given
-	// user. If the user is currently in the room, returns the room's current
-	// members, if not returns an empty array (TODO: Fix it)
-	// If the user requesting the list of members has never been in the room,
-	// returns nil.
-	// If there was an issue retrieving the events, returns an error.
-	GetMembershipEvents(roomNID types.RoomNID, requestSenderUserID string) (events []types.Event, err error)
+	// Lookup the event IDs for a batch of event numeric IDs.
+	// Returns an error if the retrieval went wrong.
+	EventIDs(eventNIDs []types.EventNID) (map[types.EventNID]string, error)
+	// Save a given room alias with the room ID it refers to.
+	// Returns an error if there was a problem talking to the database.
+	SetRoomAlias(alias string, roomID string) error
+	// Lookup the room ID a given alias refers to.
+	// Returns an error if there was a problem talking to the database.
+	GetRoomIDFromAlias(alias string) (string, error)
+	// Lookup all aliases referring to a given room ID.
+	// Returns an error if there was a problem talking to the database.
+	GetAliasesFromRoomID(roomID string) ([]string, error)
+	// Remove a given room alias.
+	// Returns an error if there was a problem talking to the database.
+	RemoveRoomAlias(alias string) error
+	// Lookup the membership of a given user in a given room.
+	// Returns the numeric ID of the latest membership event sent from this user
+	// in this room, along a boolean set to true if the user is still in this room,
+	// false if not.
```

Comment:
It looks like you've accidentally re-added the ".*Alias.*" methods here. I don't think they are needed here anymore since the aliases are handled in a separate package.